app/backend/approaches/chatreadretrieveread.py

```diff
@@ -259,21 +259,19 @@ def get_messages_from_history(
         for shot in few_shots:
             message_builder.append_message(shot.get("role"), shot.get("content"))
 
-        user_content = user_conv
         append_index = len(few_shots) + 1
 
         message_builder.append_message(self.USER, user_content, index=append_index)
 
         for h in reversed(history[:-1]):
+            if message_builder.token_length > max_tokens:
```

Comment:
Moved up to account for situation of running out of tokens with just the system prompt and user message